File: examples-objc/LocusLabsSamples/LocusLabsSamples/ViewController.m
//
//  ViewController.m
//  LocusLabsSamples
//
//  Copyright © 2020 LocusLabs. All rights reserved.
//

#import "ViewController.h"
#import "ShowFullscreenMap.h"
#import "ShowEmbeddedMap.h"
#import "BundledMap.h"
#import "CustomActions.h"
#import "DirectionsRequested.h"
#import "DirectionsStepsETA.h"
#import "DirectionsUI.h"
#import "ExternalLocationServices.h"
#import "HeadlessMode.h"
#import "MapBasics.h"
#import "Markers.h"
#import "POIButtons.h"
#import "POIShow.h"
#import "RecommendedSearches.h"
#import "SearchAutoDisplay.h"
#import "SearchCategories.h"
#import "SearchGeneral.h"
#import "SearchMultiTerm.h"
#import "SearchProximity.h"
#import "UICustomization.h"
#import "UserLocationSimulated.h"
#import "UserLocationTracking.h"
#import "VenueData.h"
#import "IntegrationGrab.h"

@interface ViewController ()

@property (nonatomic, strong) NSArray *samples;

@end

@implementation ViewController {
    
}
// *********************************************************************************************
// NB - You may need to run "pod update" before running this code
// *********************************************************************************************

#pragma mark Lifecycle

- (void)viewDidLoad {
    
    [super viewDidLoad];
    
    self.title = @"LocusLabs Samples";
    
    self.samples = @[@"Fullscreen Map", @"Embedded Map", @"Bundled Map", @"Custom Actions", @"Directions Requested", @"Directions - Steps & ETA",
                    @"Directions UI", @"External Location Services", @"GRAB Integration", @"Map Basics", @"Markers", @"POI Buttons", @"POI Show",
                    @"Recommended Searches", @"Search Auto Display", @"Search Categories", @"Search General", @"Search Multiterm", @"Search Proximity",
                    @"UI Customization", @"User Location Simulated", @"User Location Tracking", @"Venue Data"];
}

- (void)viewWillAppear:(BOOL)animated {
    
    [super viewWillAppear:animated];
    [self.navigationController setNavigationBarHidden:NO animated:animated];
}

- (void)viewWillDisappear:(BOOL)animated {
  
    [super viewWillDisappear:animated];
    [self.navigationController setNavigationBarHidden:YES animated:animated];
    
}

#pragma mark Delegates - TableView

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    
    return [self.samples count];
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {

    static NSString *CellIdentifier = @"Cell";
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:CellIdentifier];
    
    if (!cell) {
        
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:CellIdentifier];
        
    }
    
    cell.textLabel.text = self.samples[indexPath.row];
    
    return cell;
    
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    
    NSString *sample = self.samples[indexPath.row];
    UIViewController *vcToShow = nil;
    
    if ([sample isEqualToString:@"Fullscreen Map"]) vcToShow = [ShowFullscreenMap new];
    else if ([sample isEqualToString:@"Embedded Map"]) vcToShow = [ShowEmbeddedMap new];
    else if ([sample isEqualToString:@"Bundled Map"]) vcToShow = [BundledMap new];
    else if ([sample isEqualToString:@"Custom Actions"]) vcToShow = [CustomActions new];
    else if ([sample isEqualToString:@"Directions Requested"]) vcToShow = [DirectionsRequested new];
    else if ([sample isEqualToString:@"Directions - Steps & ETA"]) vcToShow = [DirectionsStepsETA new];
    else if ([sample isEqualToString:@"Directions UI"]) vcToShow = [DirectionsUI new];
    else if ([sample isEqualToString:@"External Location Services"]) vcToShow = [ExternalLocationServices new];
    else if ([sample isEqualToString:@"GRAB Integration"]) vcToShow = [IntegrationGrab new];
    else if ([sample isEqualToString:@"Map Basics"]) vcToShow = [MapBasics new];
    else if ([sample isEqualToString:@"Markers"]) vcToShow = [Markers new];
    else if ([sample isEqualToString:@"POI Buttons"]) vcToShow = [POIButtons new];
    else if ([sample isEqualToString:@"POI Show"]) vcToShow = [POIShow new];
    else if ([sample isEqualToString:@"Recommended Searches"]) vcToShow = [RecommendedSearches new];
    else if ([sample isEqualToString:@"Search Auto Display"]) vcToShow = [SearchAutoDisplay new];
    else if ([sample isEqualToString:@"Search Categories"]) vcToShow = [SearchCategories new];
    else if ([sample isEqualToString:@"Search General"]) vcToShow = [SearchGeneral new];
    else if ([sample isEqualToString:@"Search Multiterm"]) vcToShow = [SearchMultiTerm new];
    else if ([sample isEqualToString:@"Search Proximity"]) vcToShow = [SearchProximity new];
    else if ([sample isEqualToString:@"UI Customization"]) vcToShow = [UICustomization new];
    else if ([sample isEqualToString:@"User Location Simulated"]) vcToShow = [UserLocationSimulated new];
    else if ([sample isEqualToString:@"User Location Tracking"]) vcToShow = [UserLocationTracking new];
    else if ([sample isEqualToString:@"Venue Data"]) vcToShow = [VenueData new];
    
    [self.navigationController pushViewController:vcToShow animated:nil];
}

@end

File: LocusLabsSamplesSwift/LocusLabsSamplesSwift/ViewController.swift
//
//  ViewController.swift
//  LocusLabsSamplesSwift
//
//  Copyright © 2020 LocusLabs. All rights reserved.
//
// *********************************************************************************************
// NB - You may need to run "pod update" before running this code
// *********************************************************************************************

import UIKit

class ViewController: UITableViewController {

    var samples: Array<String>!
    
    // MARK: Lifecycle
    override func viewDidLoad() {
        
        super.viewDidLoad()
        title = "Demo Samples"
        
        samples = ["Fullscreen Map", "Embedded Map", "Bundled Map", "Custom Actions", "Directions Requested", "Directions - Steps & ETA",
        "Directions UI", "External Location Services", "GRAB Integration" ,"Map Basics", "Markers", "POI Buttons", "POI Show",
        "Recommended Searches", "Search Auto Display", "Search Categories", "Search General", "Search Multiterm", "Search Proximity",
        "UI Customization", "User Location Simulated", "User Location Tracking", "Venue Data"]
    }

    override func viewWillAppear(_ animated: Bool) {
            
        super.viewWillAppear(animated)
        navigationController?.setNavigationBarHidden(false, animated: animated)
    }
    
    override func viewWillDisappear(_ animated: Bool) {
            
        super.viewWillDisappear(animated)
        navigationController?.setNavigationBarHidden(true, animated: animated)
    }
    
    // MARK: Delegates - TableView
    override func numberOfSections(in tableView: UITableView) -> Int {
        
        return 1
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        
        return samples.count
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        
        let reuseIdentifier = "reuseIdentifier"
        var cell = tableView.dequeueReusableCell(withIdentifier: reuseIdentifier)
        
        if cell == nil {
            
            cell = UITableViewCell(style: .default, reuseIdentifier: reuseIdentifier)
        }
        
        let sample = samples[indexPath.row]
        cell?.textLabel?.text = sample
        
        return cell!
    }
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        
        let sample = samples[indexPath.row]
        var vcToShow: UIViewController?
        
        if sample == "Fullscreen Map" {vcToShow = ShowFullscreenMap() as UIViewController}
        else if sample == "Embedded Map" {vcToShow = ShowEmbeddedMap() as UIViewController}
        else if sample == "Bundled Map" {vcToShow = BundledMap() as UIViewController}
        else if sample == "Custom Actions" {vcToShow = CustomActions() as UIViewController}
        else if sample == "Directions Requested" {vcToShow = DirectionsRequested() as UIViewController}
        else if sample == "Directions - Steps & ETA" {vcToShow = DirectionsStepsETA() as UIViewController}
        else if sample == "Directions UI" {vcToShow = DirectionsUI() as UIViewController}
        else if sample == "External Location Services" {vcToShow = ExternalLocationServices() as UIViewController}
        else if sample == "GRAB Integration" {vcToShow = IntegrationGrab() as UIViewController}
        else if sample == "Map Basics" {vcToShow = MapBasics() as UIViewController}
        else if sample == "Markers" {vcToShow = Markers() as UIViewController}
        else if sample == "POI Buttons" {vcToShow = POIButtons() as UIViewController}
        else if sample == "POI Show" {vcToShow = POIShow() as UIViewController}
        else if sample == "Recommended Searches" {vcToShow = RecommendedSearches() as UIViewController}
        else if sample == "Search Auto Display" {vcToShow = SearchAutoDisplay() as UIViewController}
        else if sample == "Search Categories" {vcToShow = SearchCategories() as UIViewController}
        else if sample == "Search General" {vcToShow = SearchGeneral() as UIViewController}
        else if sample == "Search Multiterm" {vcToShow = SearchMultiTerm() as UIViewController}
        else if sample == "Search Proximity" {vcToShow = SearchProximity() as UIViewController}
        else if sample == "UI Customization" {vcToShow = UICustomization() as UIViewController}
        else if sample == "User Location Simulated" {vcToShow = UserLocationSimulated() as UIViewController}
        else if sample == "User Location Tracking" {vcToShow = UserPositionTracking() as UIViewController}
        else if sample == "Venue Data" {vcToShow = VenueData() as UIViewController}
        
        guard vcToShow != nil else {
            return
        }
        
        vcToShow?.modalPresentationStyle  = .fullScreen
        navigationController?.pushViewController(vcToShow!, animated: true)
    }
}

